akka-http-caching/src/main/scala/akka/http/caching/javadsl/LoadingCacheSettings.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2017-2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.caching.javadsl
+
+import akka.annotation.DoNotInherit
+import akka.http.caching.impl.settings.LoadingCacheSettingsImpl
+import akka.http.javadsl.settings.SettingsCompanion
+import com.typesafe.config.Config
+
+import scala.concurrent.duration.Duration
+
+/**
+ * Public API but not intended for subclassing
+ */
+@DoNotInherit
+abstract class LoadingCacheSettings private[http] () { self: LoadingCacheSettingsImpl =>
+
+  /* JAVA APIs */
+  def getMaxCapacity: Int
+  def getRefreshAfterWrite: Duration
+  def getExpireAfterWrite: Duration
```

Comment:
For Java should return and accept `java.time.Duration` rather than `scala.concurrent.duration.Duration` (but for Scala we want the Scala duration ofc)

Reply:
Ah, sorry, missed your comment, yes, I'm sure, that must have been an oversight for the original APIs (if you could add there as well it would be great)

Reply:
cool, I can fix that! But would that be a breaking api change for people to transition?

Reply:
Good point, we have the APIs marked with `ApiMayChange` which signals that we do not guarantee compatibility quite yet, but it's still not super nice to break things.

 I think constructing Scala Duration from Java is not very pleasant/obvious either so maybe the risk is smallish and Java devs may accept it since it improves their lives.

Reply:
👍  alright I'll fix it!